File: read/src/main/java/com/hutcwp/read/ui/home/top/WrapPhotoFragment.java
package com.hutcwp.read.ui.home.top;

import org.jetbrains.annotations.NotNull;

import androidx.fragment.app.Fragment;
import com.hutcwp.read.ui.home.sub.picture.PictureFragment;
import hut.cwp.annotations.mvp.DelegateBind;

/**
 * @ProjectName: LifeUtil$
 * @Package: com.hutcwp.lifeutil.ui.home.top$
 * @ClassName: WrapPhotoFragment$
 * @Description:
 * @Author: caiwenpeng
 * @CreateDate: 2020/8/16$ 7:13 PM$
 */
@DelegateBind(presenter = WrapPhotoPresenter.class)
public class WrapPhotoFragment extends TopFragment {

    @NotNull
    @Override
    public String getTitle() {
        return "福利";
    }

    @NotNull
    @Override
    public Fragment getFragment() {
        return new PictureFragment();
    }
}

File: read/src/main/java/com/hutcwp/read/ui/home/tabs/WrapPhotoFragment.kt
package com.hutcwp.read.ui.home.tabs

import androidx.fragment.app.Fragment
import com.hutcwp.read.entitys.ReadCategory
import com.hutcwp.read.ui.home.impl.picture.PictureFragment
import hut.cwp.annotations.mvp.DelegateBind

/**
 * @ClassName: WrapPhotoFragment$
 * @Description:
 * @Author: kevin
 * @CreateDate: 2020/8/16$ 7:13 PM$
 */
@DelegateBind(presenter = WrapPhotoPresenter::class)
class WrapPhotoFragment : TopFragment() {

    override val title: String get() = "福利"

    override fun getFragment(category: ReadCategory): Fragment {
        return PictureFragment.instance(category.url)
    }
}